examples/models/llama/attention.py

```diff
@@ -0,0 +1,135 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple, Any
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from executorch.examples.models.llama.llama_transformer import ModelArgs
+
+class Attention(nn.Module, ABC):
+    """Abstract base class for attention mechanisms with unified interface."""
+    @abstractmethod
+    def forward(
+        self,
+        x: torch.Tensor,
+        freqs_cos: torch.Tensor,
+        freqs_sin: torch.Tensor,
+        mask: Optional[torch.Tensor] = None,
+        input_pos: Optional[torch.Tensor] = None,
+        in_cache_state: Optional[Any] = None,
+        out_cache_state: Optional[Any] = None,
```

Comment:
replace them with kwargs

Reply:
directly using kwargs may break type-safe. Keep them as is and consider using TypedDict and Unpack for kwarg type checking later.